src/main/java/io/github/amerebagatelle/fabricskyboxes/skyboxes/AbstractSkybox.java

```diff
@@ -319,37 +323,29 @@ public void parseJson(JsonObjectWrapper jsonObjectWrapper) {
             }
         }
         element = jsonObjectWrapper.getOptionalValue("biomes").orElse(null);
-        if (element != null) {
-            if (element.isJsonArray()) {
-                for (JsonElement jsonElement : element.getAsJsonArray()) {
-                    biomes.add(new Identifier(jsonElement.getAsString()));
-                }
-            } else if (JsonHelper.isString(element)) {
-                biomes.add(new Identifier(element.getAsString()));
-            }
-        }
+        this.processIds(element, biomes);
         element = jsonObjectWrapper.getOptionalValue("dimensions").orElse(null);
-        if (element != null) {
-            if (element.isJsonArray()) {
-                for (JsonElement jsonElement : element.getAsJsonArray()) {
-                    dimensions.add(new Identifier(jsonElement.getAsString()));
-                }
-            } else if (JsonHelper.isString(element)) {
-                dimensions.add(new Identifier(element.getAsString()));
-            }
-        }
+        this.processIds(element, worlds);
         element = jsonObjectWrapper.getOptionalValue("heightRanges").orElse(null);
         if (element != null) {
             JsonArray array = element.getAsJsonArray();
             for (JsonElement jsonElement : array) {
                 JsonArray insideArray = jsonElement.getAsJsonArray();
                 float low = insideArray.get(0).getAsFloat();
                 float high = insideArray.get(1).getAsFloat();
-                if (high > low) {
-                    this.heightRanges.add(new HeightEntry(low, high));
-                } else {
-                    FabricSkyBoxesClient.getLogger().warn("Skybox " + getType() + " contains invalid height ranges.");
+                this.heightRanges.add(new HeightEntry(low, high));
+            }
+        }
+    }
+
+    private void processIds(JsonElement element, List<Identifier> list) {
```

Comment:
Umm, maybe I'm misunderstanding here, but wouldn't this just add the element to a copied list, not the actual list?

Reply:
This code was duplicated in some methods. Just decided to make a method for it. 
And no, lists are passed by reference, which allows for stuff like this to happen.